File: android/app/src/main/java/com/myfibonacci/java/FibonacciSeries.java
package com.myfibonacci.java;

class FibonacciSeries {
    private long num;
    private long num1;
    private long num2;

    FibonacciSeries() {
        num = 0;
        num1 = 0;
        num2 = 1;
    }

    public String getInitFibonacciNum() {
        return num1 + " " + num2;
    }

    public String getFibonacciNum() {
        return " " + num;
    }

    public void generateNextFibonacciNum() {
        num = num1 + num2;
        num1 = num2;
        num2 = num;
    }

    public void resetFibonacciNum() {
        num = 0;
        num1 = 0;
        num2 = 1;
    }
}

File: android/app/src/main/java/com/myfibonacci/kotlin/FibonacciSeries.kt
package com.myfibonacci.kotlin


class FibonacciSeries {
    private var num: Long? = null
    private var num1: Long? = null
    private var num2: Long? = null

    val initFibonacciNum: String
        get() = "${num1 ?: 0} ${num2 ?: 0}"

    val fibonacciNum: String
        get() = " ${num ?: 0}"

    fun generateNextFibonacciNum() {
        num = (num1 ?: 0) + (num2 ?: 0)
        num1 = num2
        num2 = num
    }

    fun resetFibonacciNum() {
        num = 0
        num1 = 0
        num2 = 1
    }

    init {
        num = 0
        num1 = 0
        num2 = 1
    }
}